deployer/deployer-services/src/main/java/io/hyscale/deployer/services/exception/DeployerErrorCodes.java

```diff
@@ -16,59 +16,71 @@
+    FAILED_TO_TAIL_POD("Failed to tail pod logs",HyscaleErrorGroup.GET_LOGS),
+    REPLICA_DOES_NOT_EXIT("Cannot find replica \"{}\"",HyscaleErrorGroup.GET_LOGS),
+    RESOURCE_NOT_FOUND("Resource not found {} ",HyscaleErrorGroup.DEPLOYER_APPLY),
+    FAILED_TO_GET_RESOURCE("Failed to get resource {} ",HyscaleErrorGroup.DEPLOYER_APPLY),
+    FAILED_TO_CREATE_RESOURCE("Failed to create resource {}",HyscaleErrorGroup.DEPLOYER_APPLY),
     FAILED_TO_DELETE_RESOURCE("Failed to delete resource {}"),
     FAILED_TO_UPDATE_RESOURCE("Failed to update resource {}"),
     FAILED_TO_PATCH_RESOURCE("Failed to patch resource {}"),
-    FAILED_TO_GET_SERVICE_ADDRESS("Failed to get service address"),
-    MANIFEST_REQUIRED("Manifest required"),
-    FAILED_TO_APPLY_MANIFEST("Failed to apply manifests"),
+    FAILED_TO_GET_SERVICE_ADDRESS("Failed to get service address of service \"{}\" in namespace \"{}\"",HyscaleErrorGroup.GET_SERVICE_IP),
+    MANIFEST_REQUIRED("Manifest required",HyscaleErrorGroup.DEPLOYER_APPLY),
+    FAILED_TO_APPLY_MANIFEST("Failed to apply manifests",HyscaleErrorGroup.DEPLOYER_APPLY),
     FAILED_TO_READ_MANIFEST("Failed to read manifests"),
     SERVICE_REQUIRED("Service required"),
     APPLICATION_REQUIRED("Application name required"),
-    FAILED_TO_GET_LOGS("Failed to get pod logs"),
-    FAILED_TO_INITIALIZE_POD("Failed to initialize pod"),
-    POD_FAILED_READINESS("Pod Readiness failed"),
+    FAILED_TO_GET_LOGS("Failed to get pod logs",HyscaleErrorGroup.GET_LOGS),
+    FAILED_TO_INITIALIZE_POD("Failed to initialize pod",HyscaleErrorGroup.WAIT_FOR_DEPLOYMENT),
+    POD_FAILED_READINESS("Pod Readiness failed",HyscaleErrorGroup.WAIT_FOR_DEPLOYMENT),
     POD_SELECTOR_NOT_FOUND("Pod selector not found"),
-    FAILED_TO_CREATE_POD("Failed to create pod"),
+    FAILED_TO_CREATE_POD("Failed to create pod",HyscaleErrorGroup.WAIT_FOR_DEPLOYMENT),
     OPERATION_NOT_SUPPORTED("Operation {} not supported for resource {}"),
     INVALID_STORAGE_CLASS_FOR_VOLUME("Storage class {} in your hspec is found to be invalid , allowed values from your cluster are {}"),
     NO_STORAGE_CLASS_IN_K8S("No storage class defined in your kubernetes cluster. Please contact your cluster administrator"),
     MISSING_DEFAULT_STORAGE_CLASS("Missing default storage class from the cluster so kindly define storage class in your hspec volumes [{}] , allowed values are : {} "),
     MISSING_UNIQUE_DEFAULT_STORAGE_CLASS("More than 1 default storage class found in cluster, either define single default storage class or specify a storage class in your hspec volumes [{}] , allowed values are : {} "),
     TIMEDOUT_WHILE_WAITING_FOR_SCALING("Timedout while waiting for {} to be scaled"),
     ERROR_WHILE_SCALING("Error while scaling : {}"),
-    FAILED_TO_RETRIEVE_SERVICE_REPLICAS("Failed to retrieve service replicas"),
+    FAILED_TO_RETRIEVE_SERVICE_REPLICAS("Failed to retrieve service replicas",HyscaleErrorGroup.WAIT_FOR_DEPLOYMENT),
```

Comment:
FAILED_TO_RETRIEVE_SERVICE_REPLICAS is being thrown in **watch()** method of V1PodHandler.java which is being called by **waitforDeployment()** method in KubernetedsDeployer.java. So I though it would be part of wait for deployment phase. Can you please check and confirm once.



![image](https://user-images.githubusercontent.com/66588430/84871889-bbc82980-b09e-11ea-9763-835f60e3d264.png)

![image](https://user-images.githubusercontent.com/66588430/84872077-06e23c80-b09f-11ea-91d9-fe75441b4f23.png)